imap/ical_support.c

```diff
@@ -454,6 +528,18 @@ static icaltimetype multirrule_iterator_next(struct multirrule_iterator *iter)
         }
     }
     iter->entries[min].next = icalrecur_iterator_next(iter->entries[min].icaliter);
+
+    // now forward all iterators that are at exactly the same recurid
+    for (i = 0; i < iter->nentries; i++) {
+        struct multirrule_iterator_entry *entry = iter->entries + i;
+        if (icaltime_is_null_time(entry->next)) {
+            continue;
+        }
+        if (min != i && !icaltime_compare(next, entry->next)) {
+            iter->entries[i].next = icalrecur_iterator_next(iter->entries[i].icaliter);
```

Comment:
I wonder if rather than having a second loop, this could be part of the first loop as:

```
else if (!icaltime_compare(next, entry->next)) {
            iter->entries[i].next = icalrecur_iterator_next(iter->entries[i].icaliter);
}

This would also bump duplicates of future recurrences until we find the actual minimum next
```